cmd/kube-scheduler/app/options/configfile.go

```diff
@@ -37,39 +40,61 @@ func loadConfigFromFile(file string) (*kubeschedulerconfig.KubeSchedulerConfigur
 	return loadConfig(data)
 }
 
-func loadConfig(data []byte) (*kubeschedulerconfig.KubeSchedulerConfiguration, error) {
+func loadConfig(data []byte) (*config.KubeSchedulerConfiguration, error) {
 	// The UniversalDecoder runs defaulting and returns the internal type by default.
-	obj, gvk, err := kubeschedulerscheme.Codecs.UniversalDecoder().Decode(data, nil, nil)
+	obj, gvk, err := scheme.Codecs.UniversalDecoder().Decode(data, nil, nil)
 	if err != nil {
 		return nil, err
 	}
-	if cfgObj, ok := obj.(*kubeschedulerconfig.KubeSchedulerConfiguration); ok {
+	if cfgObj, ok := obj.(*config.KubeSchedulerConfiguration); ok {
+		// We don't set this field in pkg/scheduler/apis/config/{version}/conversion.go
+		// because the field will be cleared later by API machinery during
+		// conversion. To work around that, we set the API version field of the
+		// internal object here to the version we read from file. We need this for
+		// later when we need to make decisions based on the external version.
+		cfgObj.TypeMeta.APIVersion = gvk.GroupVersion().String()
```

Comment:
do you have a pointer to the decisions made based on this field? it's pretty strange for the external version source to change the interpretation of the converted internal config

Reply:
if the only decision made is a pass/fail validation decision, that would be helpful to clarify here. we explicitly don't want the internal configuration getting interpreted differently based on version

Reply:
There are two places, the validation and the plugin config defaulting logic inside the framework. The former will continue to exist, the latter will be removed as a followup to this PR (https://github.com/kubernetes/kubernetes/issues/102211).

Reply:
*three places. We also use this in `--write-config-to`. I've added a comment in the internal KubeSchedulerConfiguration with references to exactly what places use this field.